src/middleware/immer.ts

```diff
@@ -0,0 +1,66 @@
+import { produce } from 'immer'
+import type { Draft } from 'immer'
```

Comment:
We don't emit `import type` assuming old TS doesn't support it.
```suggestion
import { Draft, produce } from 'immer'
```

Reply:
But then `eslint(import/named)` complains because it doesn't understand `Draft` is a type. As I had probably said before TypeScript projects don't even need `eslint(import/named)`. So should I remove the lint rule?

Reply:
Oh, I wasn't aware that. (Why doesn't eslint understand the `Draft` type??)
We maintain the same eslintrc json file among zustand/jotai/valtio, so for this PR, please just add `// eslint-disable-next-line import/named`.

Reply:
> Why doesn't eslint understand the Draft type??

Not sure, perhaps eslint doesn't check `.d.ts`, or it could even be a bug

> please just add // eslint-disable-next-line import/named.

Sure